baselines/fedmix/README.md

```diff
@@ -0,0 +1,128 @@
+| -- | -- | -- | -- |
+| num classes | 10 | 100 | 62 |
+| num clients | 60 | 100 | 100 |
+| num classes per client | 2 | 20 | |
+| non-iidness type | data from selected classes | data from selected classes | data from single writer |
+
+
+****Training Hyperparameters:****
+
+| Hyperparameter | CIFAR10 Experiments | CIFAR100 Experiments | FEMNIST Experiments |
+| -- | -- | -- | -- |
+| local epochs | 2 | 10 | 10 |
+| local learning rate | 0.01 | 0.01 | 0.01|
+| local lr decay after round | 0.999 | 0.999 | 0.999 |
+| local batch size | 10 | 10 | 10 |
+| num classes per client | 2 | 20 |  |
+| total number of clients | 60 | 100 | 100 |
+| clients per round | 15 | 10 | 10 |
+| mash batch size | full local data (`all`) | full local data (`all`) | full local data (`all`)|
+| mixup ratio (fedmix) | 0.05 | 0.1 | 0.2 |
+| mixup ratio (naivemix) | 0.1 | 0.1 | 0.2 |
+| num rounds | 500 | 500 | 200 |
+
+
+## Environment Setup
+
+```
+# set local python version via pyenv
+pyenv local 3.10.6
+
+# then fix that for poetry
+poetry env use 3.10.6
+
+# then install poetry env
+poetry install
+
+# activate the environment
+poetry shell
+```
+
+## Running the Experiments
+
+To run an experiment with default hyperparameters (as mentioned in the paper), execute:
+
+```
+python -m fedmix.main +experiment={dataset}_{strategy}
+# dataset can be: cifar10, cifar100, femnist
+# strategy can be: fedavg, naivemix, fedmix
+```
+
+To run custom experiments: create a new `.yaml` file in `conf/experiment` and override the default hyperparameters.
+
+## Expected Results
+
+To get all results, run:
+
+```bash
+# femnist experiments
+python -m fedmix.main --multirun +experiment=femnist_fedavg,femnist_naivemix,femnist_fedmix
+
+# cifar10 experiments
+python -m fedmix.main --multirun +experiment=cifar10_fedavg,cifar10_naivemix,cifar10_fedmix
+
+# cifar100 experiments
+python -m fedmix.main --multirun +experiment=cifar100_fedavg,cifar100_naivemix,cifar100_fedmix
+```
+
+To generate table (similar to `Table 1` of paper) and figures (similar to `Figure 2` of paper), run the above commands and then:
+
+```bash
+# table
+python -m fedmix.generate_media --output_type=table --output_path="_static/table.png" --input_directory="results/"
+
+# femnist learning curve
+python -m fedmix.generate_media --output_type=figure --output_path="_static/femnist.png" --input_directory="results/" --dataset_name="femnist"
+
+# cifar10 learning curve
+python -m fedmix.generate_media --output_type=figure --output_path="_static/cifar10.png" --input_directory="results/" --dataset_name="cifar10"
+
+# cifar100 learning curve
+python -m fedmix.generate_media --output_type=figure --output_path="_static/cifar100.png" --input_directory="results/" --dataset_name="cifar100"
+```
+
+The expected results are shown below:
+
+| Algorithm | FEMNIST | CIFAR10 | CIFAR100 |
+| -- | -- | -- | -- |
+| FedAvg | **82.49** (85.3) | 62.77 (73.8) | 27.47 (50.4) |
```

Comment:
I got very different results for femnist-fedavg with the code as in this PR. (~65%)